### PR TITLE
FA-94 createAccountCore feature module created

### PR DESCRIPTION
**Pull request by HamdanSCode**

### CODE REVIEW

File: app/src/main/java/com/hamdan/forzenbook/view/login/CreateAccount.kt [hunk 1]
```diff
@@ -130,9 +130,9 @@ private fun Content(
         onNext = { focusManager.moveFocus(FocusDirection.Down) }
     )
     if (firstName.isNotEmpty() && !stateFirstName.error.isValid()) {
-        if (stateFirstName.error == LoginError.NameError.Length) {
+        if (stateFirstName.error == com.hamdan.forzenbook.core.LoginError.NameError.Length) {
```
Comment (ForzenAcademy) [inline app/src/main/java/com/hamdan/forzenbook/view/login/CreateAccount.kt]: Import this so you don't have these giant package names on classes

File: app/src/main/java/com/hamdan/forzenbook/view/login/MainLogin.kt [hunk 2]
```diff
@@ -105,9 +103,9 @@ private fun Content(
         },
     )
     if (email.isNotEmpty() && !emailError.isValid()) {
-        if (emailError == LoginError.EmailError.Length) {
+        if (emailError == com.hamdan.forzenbook.core.LoginError.EmailError.Length) {
```
Comment (ForzenAcademy) [inline app/src/main/java/com/hamdan/forzenbook/view/login/MainLogin.kt]: Fix

File: core/build.gradle [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
```
Comment (ForzenAcademy) [inline core/build.gradle]: You should be on Java 11

File: core/build.gradle [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (ForzenAcademy) [inline core/build.gradle]: Java 11

File: core/proguard-rules.pro [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (ForzenAcademy) [inline core/proguard-rules.pro]: Delete empty files like this and `consumer-rules.pro`

File: core/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ForzenAcademy) [inline core/src/main/AndroidManifest.xml:1]: Condense this file

File: core/src/main/java/com/hamdan/forzenbook/core/GlobalConstants.kt [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package com.hamdan.forzenbook.core
+
+object GlobalConstants {
+    const val EMAIL_LENGTH_LIMIT = 30
```
Comment (ForzenAcademy) [inline core/src/main/java/com/hamdan/forzenbook/core/GlobalConstants.kt]: This got updated. You ignore Sergei!?

File: createAccountCore/build.gradle [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
```
Comment (ForzenAcademy) [inline createAccountCore/build.gradle]: Java 11

File: createAccountCore/build.gradle [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
```
Comment (ForzenAcademy) [inline createAccountCore/build.gradle]: 11

File: createAccountCore/proguard-rules.pro [hunk 10]
```diff
@@ -0,0 +1,21 @@
+# Add project specific ProGuard rules here.
```
Comment (ForzenAcademy) [inline createAccountCore/proguard-rules.pro]: Erase empty files

File: createAccountCore/src/main/AndroidManifest.xml [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ForzenAcademy) [inline createAccountCore/src/main/AndroidManifest.xml:1]: Condense

File: createAccountCore/src/main/java/com/hamdan/forzenbook/createaccountcore/data/network/CreateAccountService.kt [hunk 12]
```diff
@@ -1,4 +1,4 @@
-package com.hamdan.forzenbook.data.network
+package com.hamdan.forzenbook.createaccountcore.data.network
```
Comment (ForzenAcademy) [inline createAccountCore/src/main/java/com/hamdan/forzenbook/createaccountcore/data/network/CreateAccountService.kt]: Package should be `createaccount.core`

File: createAccountCore/src/main/java/com/hamdan/forzenbook/createaccountcore/di/CreateAccountModule.kt [hunk 13]
```diff
@@ -55,4 +54,5 @@ object CreateAccountModule {
     }
 
     private const val MODULE_NAME = "create_account"
+    private const val LOGIN_BASE_URL = "https://forzen.dev/api/"
```
Comment (ForzenAcademy) [inline createAccountCore/src/main/java/com/hamdan/forzenbook/createaccountcore/di/CreateAccountModule.kt]: Do you have duplicates of this?

File: createAccountCore/src/test/java/com/hamdan/forzenbook/createaccountcore/domain/DomainUnitTest.kt [hunk 14]
```diff
@@ -0,0 +1,100 @@
+package com.hamdan.forzenbook.createaccountcore.domain
```
Comment (ForzenAcademy) [inline createAccountCore/src/test/java/com/hamdan/forzenbook/createaccountcore/domain/DomainUnitTest.kt]: The bin and txt files you have in the changed file list under this file need to not be checked in

File: core/src/main/java/com/hamdan/forzenbook/core/Entry.kt [hunk 15]
```diff
@@ -0,0 +1,6 @@
+package com.hamdan.forzenbook.core
+
+data class Entry(
```
Comment (ForzenAcademy) [inline core/src/main/java/com/hamdan/forzenbook/core/Entry.kt:3]: This name is generic, `Entry`, but then it includes `LoginError` which isn't generic. And is this used JUST for login errors? If this is to be reusable don't tie it tight to Login error

File: app/src/main/java/com/hamdan/forzenbook/view/login/CreateAccount.kt (hunk 1 above)
Comment (theflag505) [reply app/src/main/java/com/hamdan/forzenbook/view/login/CreateAccount.kt:133]: yea, I believe I am in all cases, but it auto put that ahead on everything, like all spots like this are greyed out.

Comment (HamdanSCode) [reply app/src/main/java/com/hamdan/forzenbook/view/login/CreateAccount.kt]: I thought it had auto changed these and didn't notice this, updated.

File: core/src/main/java/com/hamdan/forzenbook/core/GlobalConstants.kt (hunk 7 above)
Comment (HamdanSCode) [reply core/src/main/java/com/hamdan/forzenbook/core/GlobalConstants.kt]: updated comrade

File: createAccountCore/src/main/java/com/hamdan/forzenbook/createaccountcore/di/CreateAccountModule.kt (hunk 13 above)
Comment (HamdanSCode) [reply createAccountCore/src/main/java/com/hamdan/forzenbook/createaccountcore/di/CreateAccountModule.kt]: Yes I do, I was thinking for some reason this shouldn't go in the global constants, but now thinking about it again why shouldn't it.

File: core/src/main/java/com/hamdan/forzenbook/core/Entry.kt (hunk 15 above)
Comment (HamdanSCode) [reply core/src/main/java/com/hamdan/forzenbook/core/Entry.kt:3]: Made a generic Error interface which LoginError now implements, made the data class Entry have its error field changed of type LoginError to Error.

File: createAccountCore/src/test/java/com/hamdan/forzenbook/createaccountcore/domain/DomainUnitTest.kt (hunk 14 above)
Comment (HamdanSCode) [reply createAccountCore/src/test/java/com/hamdan/forzenbook/createaccountcore/domain/DomainUnitTest.kt]: removed them

**Review by ForzenAcademy — APPROVED** [review]

**Pull request merged**